src/grid.tsx

```diff
@@ -0,0 +1,209 @@
+import {assignInlineVars} from '@vanilla-extract/dynamic';
+import classNames from 'classnames';
+import * as React from 'react';
+import * as styles from './grid.css';
+import {getPrefixedDataAttributes} from './utils/dom';
+
+import type {DataAttributes} from './utils/types';
+
+type RowsColumns = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
+type Gap = 0 | 2 | 4 | 8 | 12 | 16 | 24 | 32 | 40 | 48 | 56 | 64 | 72 | 80;
```

Comment:
According to CSS docs, gap can be also a percentage ([Link](https://developer.mozilla.org/en-US/docs/Web/CSS/gap))

Reply:
yes, and according to CSS docs it can be 3 or 5 too :P but I think we don't want to support those values in mistica. But @aweell, if you find supporting % useful for mistica I can add support

Reply:
Why not supporting %? I think it's not rare for an user to use gap with percentages.

Reply:
I see gap here equivalent to the `space` prop in the Inline or the Stack components, if percentage values weren't needed there I wouldn't add support until the need arises.